pageserver/src/tenant/layer_map.rs

```diff
@@ -147,43 +146,22 @@ impl Drop for BatchedUpdates<'_> {
 }
 
 /// Return value of LayerMap::search
-#[derive(Eq, PartialEq, Debug)]
+#[derive(Eq, PartialEq, Debug, Hash)]
 pub struct SearchResult {
     pub layer: Arc<PersistentLayerDesc>,
     pub lsn_floor: Lsn,
 }
 
-pub struct OrderedSearchResult(SearchResult);
-
-impl Ord for OrderedSearchResult {
-    fn cmp(&self, other: &Self) -> Ordering {
-        self.0.lsn_floor.cmp(&other.0.lsn_floor)
-    }
-}
-
-impl PartialOrd for OrderedSearchResult {
-    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
-        Some(self.cmp(other))
-    }
-}
-
-impl PartialEq for OrderedSearchResult {
-    fn eq(&self, other: &Self) -> bool {
-        self.0.lsn_floor == other.0.lsn_floor
-    }
-}
-
-impl Eq for OrderedSearchResult {}
-
+#[derive(Debug)]
 pub struct RangeSearchResult {
-    pub found: BTreeMap<OrderedSearchResult, KeySpaceAccum>,
+    pub found: HashMap<SearchResult, KeySpaceAccum>,
     pub not_found: KeySpaceAccum,
```

Comment:
1. Missing a doc comment, explaining what exactly is contained in the KeySpaceAccum. Wasn't part of the origianl review that added the range search. The comment should clarify that the KeySpaceAccum is just the overlap of the SearchResult with the requested range search key range.

2. I'm curious why the switch from BTreeMap to HashMap? Just so you can remove the ordered results, or, something else?

Reply:
1: https://github.com/neondatabase/neon/pull/6576/commits/e46090dee699e0a4db125f82d6d4b71a42105378
2: See commit message